<h1>Mohamed Salem Ali</h1>
<h2>Regression Models with Keras</h2>

### Read and Clean Dataset

In [3]:
import pandas as pd
import numpy as np

In [4]:
concrete_data = pd.read_csv('concrete_data.csv')
concrete_data.head()

Cement  Blast Furnace Slag  Fly Ash  Water  Superplasticizer  \
0   540.0                 0.0      0.0  162.0               2.5   
1   540.0                 0.0      0.0  162.0               2.5   
2   332.5               142.5      0.0  228.0               0.0   
3   332.5               142.5      0.0  228.0               0.0   
4   198.6               132.4      0.0  192.0               0.0   

   Coarse Aggregate  Fine Aggregate  Age  Strength  
0            1040.0           676.0   28     79.99  
1            1055.0           676.0   28     61.89  
2             932.0           594.0  270     40.27  
3             932.0           594.0  365     41.05  
4             978.4           825.5  360     44.30

In [5]:
concrete_data.shape

(1030, 9)

In [6]:
concrete_data.describe()

Cement  Blast Furnace Slag      Fly Ash        Water  \
count  1030.000000         1030.000000  1030.000000  1030.000000   
mean    281.167864           73.895825    54.188350   181.567282   
std     104.506364           86.279342    63.997004    21.354219   
min     102.000000            0.000000     0.000000   121.800000   
25%     192.375000            0.000000     0.000000   164.900000   
50%     272.900000           22.000000     0.000000   185.000000   
75%     350.000000          142.950000   118.300000   192.000000   
max     540.000000          359.400000   200.100000   247.000000   

       Superplasticizer  Coarse Aggregate  Fine Aggregate          Age  \
count       1030.000000       1030.000000     1030.000000  1030.000000   
mean           6.204660        972.918932      773.580485    45.662136   
std            5.973841         77.753954       80.175980    63.169912   
min            0.000000        801.000000      594.000000     1.000000   
25%            0.000000        932.000000      730.950000     7.000000   
50%            6.400000        968.000000      779.500000    28.000000   
75%           10.200000       1029.400000      824.000000    56.000000   
max           32.200000       1145.000000      992.600000   365.000000   

          Strength  
count  1030.000000  
mean     35.817961  
std      16.705742  
min       2.330000  
25%      23.710000  
50%      34.445000  
75%      46.135000  
max      82.600000

concrete_data.isnull().sum()

### Split data into predictors and target

In [7]:
concrete_data_columns = concrete_data.columns

predictors = concrete_data[concrete_data_columns[concrete_data_columns != 'Strength']] # all columns except Strength
target = concrete_data['Strength'] # Strength column

In [8]:
predictors.head()

Cement  Blast Furnace Slag  Fly Ash  Water  Superplasticizer  \
0   540.0                 0.0      0.0  162.0               2.5   
1   540.0                 0.0      0.0  162.0               2.5   
2   332.5               142.5      0.0  228.0               0.0   
3   332.5               142.5      0.0  228.0               0.0   
4   198.6               132.4      0.0  192.0               0.0   

   Coarse Aggregate  Fine Aggregate  Age  
0            1040.0           676.0   28  
1            1055.0           676.0   28  
2             932.0           594.0  270  
3             932.0           594.0  365  
4             978.4           825.5  360

In [9]:
target.head()

0    79.99
1    61.89
2    40.27
3    41.05
4    44.30
Name: Strength, dtype: float64

### normalize the data by substracting the mean and dividing by the standard deviation.

In [10]:
predictors_norm = (predictors - predictors.mean()) / predictors.std()
predictors_norm.head()

Cement  Blast Furnace Slag   Fly Ash     Water  Superplasticizer  \
0  2.476712           -0.856472 -0.846733 -0.916319         -0.620147   
1  2.476712           -0.856472 -0.846733 -0.916319         -0.620147   
2  0.491187            0.795140 -0.846733  2.174405         -1.038638   
3  0.491187            0.795140 -0.846733  2.174405         -1.038638   
4 -0.790075            0.678079 -0.846733  0.488555         -1.038638   

   Coarse Aggregate  Fine Aggregate       Age  
0          0.862735       -1.217079 -0.279597  
1          1.055651       -1.217079 -0.279597  
2         -0.526262       -2.239829  3.551340  
3         -0.526262       -2.239829  5.055221  
4          0.070492        0.647569  4.976069

In [11]:
cols = predictors_norm.shape[1] # number of predictors
cols

8

## Import Keras


In [12]:
! pip install -U scikit-learn

Defaulting to user installation because normal site-packages is not writeable


In [13]:
import keras
from keras.models import Sequential
from keras.layers import Dense
import sklearn

## Build a Neural Network


In [48]:
# define regression model
def regression_model():
    # create model
    model = Sequential()
    model.add(Dense(10, activation='relu', input_shape=(cols,)))
    model.add(Dense(10, activation='relu', input_shape=(cols,)))
    model.add(Dense(10, activation='relu', input_shape=(cols,)))
    model.add(Dense(1))
    
    # compile model
    model.compile(optimizer='adam', loss='mean_squared_error')
    return model

In [49]:
# build the model
model = regression_model()

### split the data into a training and test sets

In [53]:
import numpy as np
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(predictors_norm, target, test_size=0.33, random_state=42)

In [54]:
len(X_train)

690

In [55]:
len(X_test)

340

In [56]:
# fit the model
model.fit(X_train, y_train, validation_split=0.3, epochs=50, verbose=2)

Epoch 1/50
16/16 - 1s - loss: 1624.7433 - val_loss: 1486.5360 - 1s/epoch - 84ms/step
Epoch 2/50
16/16 - 0s - loss: 1608.9061 - val_loss: 1472.0022 - 71ms/epoch - 4ms/step
Epoch 3/50
16/16 - 0s - loss: 1591.6635 - val_loss: 1454.0526 - 71ms/epoch - 4ms/step
Epoch 4/50
16/16 - 0s - loss: 1568.8342 - val_loss: 1431.2677 - 83ms/epoch - 5ms/step
Epoch 5/50
16/16 - 0s - loss: 1538.5638 - val_loss: 1400.2804 - 77ms/epoch - 5ms/step
Epoch 6/50
16/16 - 0s - loss: 1498.7596 - val_loss: 1357.6388 - 80ms/epoch - 5ms/step
Epoch 7/50
16/16 - 0s - loss: 1444.7059 - val_loss: 1298.6349 - 78ms/epoch - 5ms/step
Epoch 8/50
16/16 - 0s - loss: 1369.3561 - val_loss: 1224.2657 - 70ms/epoch - 4ms/step
Epoch 9/50
16/16 - 0s - loss: 1274.5961 - val_loss: 1128.1467 - 71ms/epoch - 4ms/step
Epoch 10/50
16/16 - 0s - loss: 1156.1508 - val_loss: 1007.9267 - 81ms/epoch - 5ms/step
Epoch 11/50
16/16 - 0s - loss: 1018.7496 - val_loss: 872.3102 - 75ms/epoch - 5ms/step
Epoch 12/50
16/16 - 0s - loss: 864.9731 - val_loss: 72

In [57]:
y_pred =model.predict(X_test)

11/11 [==============================] - 0s 2ms/step


### mean_squared_error

In [58]:
from sklearn.metrics import mean_squared_error
mean_squared_error(y_test, y_pred)

151.86105851063132

## epochs 50
### 50 mean squared errors

In [59]:
x=0
MSE_list =[]
while(x<50):
    X_train, X_test, y_train, y_test = train_test_split(predictors, target, test_size=0.33, random_state=42)
    model.fit(X_train, y_train, validation_split=0.3, epochs=50, verbose=2)
    y_pred =model.predict(X_test)
    MSE_list.append(mean_squared_error(y_test, y_pred))
    x=x+1

Epoch 1/50
16/16 - 0s - loss: 39841776.0000 - val_loss: 12858665.0000 - 135ms/epoch - 8ms/step
Epoch 2/50
16/16 - 0s - loss: 7261702.0000 - val_loss: 3458178.5000 - 73ms/epoch - 5ms/step
Epoch 3/50
16/16 - 0s - loss: 2461568.0000 - val_loss: 1641876.2500 - 75ms/epoch - 5ms/step
Epoch 4/50
16/16 - 0s - loss: 1332014.5000 - val_loss: 1035181.0625 - 72ms/epoch - 4ms/step
Epoch 5/50
16/16 - 0s - loss: 890339.1250 - val_loss: 737250.6250 - 87ms/epoch - 5ms/step
Epoch 6/50
16/16 - 0s - loss: 655929.6875 - val_loss: 557420.6250 - 75ms/epoch - 5ms/step
Epoch 7/50
16/16 - 0s - loss: 494610.3125 - val_loss: 414726.0000 - 73ms/epoch - 5ms/step
Epoch 8/50
16/16 - 0s - loss: 352724.8125 - val_loss: 275264.2188 - 86ms/epoch - 5ms/step
Epoch 9/50
16/16 - 0s - loss: 218117.1719 - val_loss: 150340.4062 - 84ms/epoch - 5ms/step
Epoch 10/50
16/16 - 0s - loss: 110391.5547 - val_loss: 65502.9141 - 74ms/epoch - 5ms/step
Epoch 11/50
16/16 - 0s - loss: 49043.6094 - val_loss: 29536.5391 - 98ms/epoch - 6ms/step


Epoch 45/50
16/16 - 0s - loss: 890.9996 - val_loss: 849.4258 - 72ms/epoch - 4ms/step
Epoch 46/50
16/16 - 0s - loss: 885.9129 - val_loss: 843.3749 - 78ms/epoch - 5ms/step
Epoch 47/50
16/16 - 0s - loss: 876.9105 - val_loss: 836.4889 - 78ms/epoch - 5ms/step
Epoch 48/50
16/16 - 0s - loss: 868.5938 - val_loss: 829.1547 - 76ms/epoch - 5ms/step
Epoch 49/50
16/16 - 0s - loss: 861.1402 - val_loss: 820.2162 - 72ms/epoch - 4ms/step
Epoch 50/50
16/16 - 0s - loss: 855.0528 - val_loss: 815.4929 - 80ms/epoch - 5ms/step
11/11 [==============================] - 0s 1ms/step
Epoch 1/50
16/16 - 0s - loss: 846.6646 - val_loss: 804.6722 - 140ms/epoch - 9ms/step
Epoch 2/50
16/16 - 0s - loss: 846.0403 - val_loss: 797.6827 - 75ms/epoch - 5ms/step
Epoch 3/50
16/16 - 0s - loss: 833.0189 - val_loss: 791.1591 - 82ms/epoch - 5ms/step
Epoch 4/50
16/16 - 0s - loss: 828.1266 - val_loss: 784.6517 - 70ms/epoch - 4ms/step
Epoch 5/50
16/16 - 0s - loss: 821.7320 - val_loss: 778.6759 - 78ms/epoch - 5ms/step
Epoch 6/50
16/16

Epoch 41/50
16/16 - 0s - loss: 540.3089 - val_loss: 463.3264 - 69ms/epoch - 4ms/step
Epoch 42/50
16/16 - 0s - loss: 538.6563 - val_loss: 457.8485 - 81ms/epoch - 5ms/step
Epoch 43/50
16/16 - 0s - loss: 533.5797 - val_loss: 456.1360 - 82ms/epoch - 5ms/step
Epoch 44/50
16/16 - 0s - loss: 541.2192 - val_loss: 453.8901 - 79ms/epoch - 5ms/step
Epoch 45/50
16/16 - 0s - loss: 529.8220 - val_loss: 451.9142 - 71ms/epoch - 4ms/step
Epoch 46/50
16/16 - 0s - loss: 528.4257 - val_loss: 449.6249 - 75ms/epoch - 5ms/step
Epoch 47/50
16/16 - 0s - loss: 526.5056 - val_loss: 451.6726 - 95ms/epoch - 6ms/step
Epoch 48/50
16/16 - 0s - loss: 522.2851 - val_loss: 446.6507 - 114ms/epoch - 7ms/step
Epoch 49/50
16/16 - 0s - loss: 522.9156 - val_loss: 445.0555 - 75ms/epoch - 5ms/step
Epoch 50/50
16/16 - 0s - loss: 519.0791 - val_loss: 445.5535 - 83ms/epoch - 5ms/step
11/11 [==============================] - 0s 2ms/step
Epoch 1/50
16/16 - 0s - loss: 515.3176 - val_loss: 440.7641 - 141ms/epoch - 9ms/step
Epoch 2/50


Epoch 37/50
16/16 - 0s - loss: 356.8851 - val_loss: 287.7594 - 65ms/epoch - 4ms/step
Epoch 38/50
16/16 - 0s - loss: 351.8175 - val_loss: 287.1342 - 66ms/epoch - 4ms/step
Epoch 39/50
16/16 - 0s - loss: 350.8061 - val_loss: 284.6005 - 67ms/epoch - 4ms/step
Epoch 40/50
16/16 - 0s - loss: 347.7603 - val_loss: 283.8111 - 65ms/epoch - 4ms/step
Epoch 41/50
16/16 - 0s - loss: 345.9499 - val_loss: 281.8937 - 71ms/epoch - 4ms/step
Epoch 42/50
16/16 - 0s - loss: 343.8212 - val_loss: 279.9214 - 100ms/epoch - 6ms/step
Epoch 43/50
16/16 - 0s - loss: 342.5010 - val_loss: 278.1056 - 75ms/epoch - 5ms/step
Epoch 44/50
16/16 - 0s - loss: 339.8959 - val_loss: 277.3648 - 74ms/epoch - 5ms/step
Epoch 45/50
16/16 - 0s - loss: 339.3382 - val_loss: 275.3869 - 84ms/epoch - 5ms/step
Epoch 46/50
16/16 - 0s - loss: 343.9889 - val_loss: 274.5689 - 69ms/epoch - 4ms/step
Epoch 47/50
16/16 - 0s - loss: 334.4193 - val_loss: 272.6573 - 72ms/epoch - 5ms/step
Epoch 48/50
16/16 - 0s - loss: 334.1863 - val_loss: 273.7809 - 7

Epoch 33/50
16/16 - 0s - loss: 213.9561 - val_loss: 171.3590 - 82ms/epoch - 5ms/step
Epoch 34/50
16/16 - 0s - loss: 213.1899 - val_loss: 171.9136 - 81ms/epoch - 5ms/step
Epoch 35/50
16/16 - 0s - loss: 210.7624 - val_loss: 171.4758 - 93ms/epoch - 6ms/step
Epoch 36/50
16/16 - 0s - loss: 209.2949 - val_loss: 170.3956 - 80ms/epoch - 5ms/step
Epoch 37/50
16/16 - 0s - loss: 209.9707 - val_loss: 170.5160 - 80ms/epoch - 5ms/step
Epoch 38/50
16/16 - 0s - loss: 208.3419 - val_loss: 170.2025 - 99ms/epoch - 6ms/step
Epoch 39/50
16/16 - 0s - loss: 209.2919 - val_loss: 170.7898 - 78ms/epoch - 5ms/step
Epoch 40/50
16/16 - 0s - loss: 209.0742 - val_loss: 168.9564 - 69ms/epoch - 4ms/step
Epoch 41/50
16/16 - 0s - loss: 210.9999 - val_loss: 175.6044 - 75ms/epoch - 5ms/step
Epoch 42/50
16/16 - 0s - loss: 210.5775 - val_loss: 170.5466 - 77ms/epoch - 5ms/step
Epoch 43/50
16/16 - 0s - loss: 211.9342 - val_loss: 170.4312 - 71ms/epoch - 4ms/step
Epoch 44/50
16/16 - 0s - loss: 203.4875 - val_loss: 169.9140 - 68

Epoch 29/50
16/16 - 0s - loss: 170.4446 - val_loss: 145.7865 - 75ms/epoch - 5ms/step
Epoch 30/50
16/16 - 0s - loss: 170.9848 - val_loss: 147.4152 - 72ms/epoch - 4ms/step
Epoch 31/50
16/16 - 0s - loss: 170.1196 - val_loss: 143.6104 - 77ms/epoch - 5ms/step
Epoch 32/50
16/16 - 0s - loss: 168.7868 - val_loss: 147.0750 - 83ms/epoch - 5ms/step
Epoch 33/50
16/16 - 0s - loss: 190.6311 - val_loss: 162.3136 - 86ms/epoch - 5ms/step
Epoch 34/50
16/16 - 0s - loss: 179.4135 - val_loss: 157.9546 - 73ms/epoch - 5ms/step
Epoch 35/50
16/16 - 0s - loss: 179.9129 - val_loss: 143.5861 - 87ms/epoch - 5ms/step
Epoch 36/50
16/16 - 0s - loss: 178.9122 - val_loss: 145.0503 - 73ms/epoch - 5ms/step
Epoch 37/50
16/16 - 0s - loss: 180.7413 - val_loss: 171.7806 - 65ms/epoch - 4ms/step
Epoch 38/50
16/16 - 0s - loss: 180.0349 - val_loss: 142.7876 - 71ms/epoch - 4ms/step
Epoch 39/50
16/16 - 0s - loss: 167.1483 - val_loss: 141.8357 - 71ms/epoch - 4ms/step
Epoch 40/50
16/16 - 0s - loss: 172.2141 - val_loss: 147.8979 - 85

Epoch 25/50
16/16 - 0s - loss: 152.8311 - val_loss: 134.4497 - 77ms/epoch - 5ms/step
Epoch 26/50
16/16 - 0s - loss: 150.7033 - val_loss: 142.0517 - 106ms/epoch - 7ms/step
Epoch 27/50
16/16 - 0s - loss: 165.8110 - val_loss: 134.7503 - 67ms/epoch - 4ms/step
Epoch 28/50
16/16 - 0s - loss: 151.1080 - val_loss: 140.1274 - 73ms/epoch - 5ms/step
Epoch 29/50
16/16 - 0s - loss: 161.8906 - val_loss: 134.1147 - 94ms/epoch - 6ms/step
Epoch 30/50
16/16 - 0s - loss: 148.7849 - val_loss: 139.0407 - 74ms/epoch - 5ms/step
Epoch 31/50
16/16 - 0s - loss: 158.8629 - val_loss: 136.0968 - 70ms/epoch - 4ms/step
Epoch 32/50
16/16 - 0s - loss: 152.7576 - val_loss: 138.9861 - 71ms/epoch - 4ms/step
Epoch 33/50
16/16 - 0s - loss: 149.1905 - val_loss: 140.5642 - 80ms/epoch - 5ms/step
Epoch 34/50
16/16 - 0s - loss: 149.8101 - val_loss: 134.9077 - 86ms/epoch - 5ms/step
Epoch 35/50
16/16 - 0s - loss: 149.5581 - val_loss: 135.3586 - 85ms/epoch - 5ms/step
Epoch 36/50
16/16 - 0s - loss: 146.6953 - val_loss: 133.7233 - 7

Epoch 21/50
16/16 - 0s - loss: 133.7497 - val_loss: 125.3974 - 71ms/epoch - 4ms/step
Epoch 22/50
16/16 - 0s - loss: 132.7412 - val_loss: 124.8550 - 64ms/epoch - 4ms/step
Epoch 23/50
16/16 - 0s - loss: 132.4219 - val_loss: 126.1216 - 66ms/epoch - 4ms/step
Epoch 24/50
16/16 - 0s - loss: 134.4850 - val_loss: 123.7182 - 67ms/epoch - 4ms/step
Epoch 25/50
16/16 - 0s - loss: 132.2132 - val_loss: 124.5091 - 68ms/epoch - 4ms/step
Epoch 26/50
16/16 - 0s - loss: 132.5711 - val_loss: 126.7026 - 70ms/epoch - 4ms/step
Epoch 27/50
16/16 - 0s - loss: 136.5598 - val_loss: 124.9611 - 75ms/epoch - 5ms/step
Epoch 28/50
16/16 - 0s - loss: 134.0708 - val_loss: 124.2671 - 80ms/epoch - 5ms/step
Epoch 29/50
16/16 - 0s - loss: 130.3319 - val_loss: 154.5650 - 80ms/epoch - 5ms/step
Epoch 30/50
16/16 - 0s - loss: 155.5636 - val_loss: 127.5104 - 72ms/epoch - 5ms/step
Epoch 31/50
16/16 - 0s - loss: 131.0300 - val_loss: 122.0973 - 90ms/epoch - 6ms/step
Epoch 32/50
16/16 - 0s - loss: 131.8531 - val_loss: 125.3786 - 99

Epoch 17/50
16/16 - 0s - loss: 131.1678 - val_loss: 158.7246 - 83ms/epoch - 5ms/step
Epoch 18/50
16/16 - 0s - loss: 135.1531 - val_loss: 142.5184 - 69ms/epoch - 4ms/step
Epoch 19/50
16/16 - 0s - loss: 135.2744 - val_loss: 111.8505 - 79ms/epoch - 5ms/step
Epoch 20/50
16/16 - 0s - loss: 115.2128 - val_loss: 124.8455 - 75ms/epoch - 5ms/step
Epoch 21/50
16/16 - 0s - loss: 121.2941 - val_loss: 109.9380 - 96ms/epoch - 6ms/step
Epoch 22/50
16/16 - 0s - loss: 116.1760 - val_loss: 109.7773 - 72ms/epoch - 5ms/step
Epoch 23/50
16/16 - 0s - loss: 118.9242 - val_loss: 117.2306 - 76ms/epoch - 5ms/step
Epoch 24/50
16/16 - 0s - loss: 125.5960 - val_loss: 109.7134 - 73ms/epoch - 5ms/step
Epoch 25/50
16/16 - 0s - loss: 114.5199 - val_loss: 114.7842 - 72ms/epoch - 5ms/step
Epoch 26/50
16/16 - 0s - loss: 149.1025 - val_loss: 141.0426 - 80ms/epoch - 5ms/step
Epoch 27/50
16/16 - 0s - loss: 124.4008 - val_loss: 113.7393 - 74ms/epoch - 5ms/step
Epoch 28/50
16/16 - 0s - loss: 122.1680 - val_loss: 122.0372 - 73

Epoch 13/50
16/16 - 0s - loss: 117.3049 - val_loss: 112.9738 - 82ms/epoch - 5ms/step
Epoch 14/50
16/16 - 0s - loss: 112.6812 - val_loss: 102.3812 - 85ms/epoch - 5ms/step
Epoch 15/50
16/16 - 0s - loss: 109.1161 - val_loss: 130.9804 - 74ms/epoch - 5ms/step
Epoch 16/50
16/16 - 0s - loss: 135.6602 - val_loss: 105.4280 - 72ms/epoch - 5ms/step
Epoch 17/50
16/16 - 0s - loss: 128.0543 - val_loss: 129.7840 - 68ms/epoch - 4ms/step
Epoch 18/50
16/16 - 0s - loss: 118.9139 - val_loss: 102.9955 - 72ms/epoch - 4ms/step
Epoch 19/50
16/16 - 0s - loss: 113.8495 - val_loss: 106.1636 - 78ms/epoch - 5ms/step
Epoch 20/50
16/16 - 0s - loss: 113.7598 - val_loss: 106.4234 - 92ms/epoch - 6ms/step
Epoch 21/50
16/16 - 0s - loss: 110.7562 - val_loss: 104.1771 - 86ms/epoch - 5ms/step
Epoch 22/50
16/16 - 0s - loss: 108.1110 - val_loss: 112.9702 - 86ms/epoch - 5ms/step
Epoch 23/50
16/16 - 0s - loss: 110.7390 - val_loss: 109.5966 - 78ms/epoch - 5ms/step
Epoch 24/50
16/16 - 0s - loss: 108.6212 - val_loss: 104.2971 - 79

Epoch 9/50
16/16 - 0s - loss: 127.0220 - val_loss: 107.9663 - 101ms/epoch - 6ms/step
Epoch 10/50
16/16 - 0s - loss: 122.1618 - val_loss: 120.8602 - 87ms/epoch - 5ms/step
Epoch 11/50
16/16 - 0s - loss: 108.0425 - val_loss: 122.4666 - 91ms/epoch - 6ms/step
Epoch 12/50
16/16 - 0s - loss: 125.6738 - val_loss: 106.7231 - 90ms/epoch - 6ms/step
Epoch 13/50
16/16 - 0s - loss: 105.7503 - val_loss: 105.5866 - 74ms/epoch - 5ms/step
Epoch 14/50
16/16 - 0s - loss: 113.5057 - val_loss: 113.6803 - 75ms/epoch - 5ms/step
Epoch 15/50
16/16 - 0s - loss: 128.4832 - val_loss: 110.1100 - 80ms/epoch - 5ms/step
Epoch 16/50
16/16 - 0s - loss: 112.2616 - val_loss: 127.5918 - 69ms/epoch - 4ms/step
Epoch 17/50
16/16 - 0s - loss: 114.7171 - val_loss: 106.2220 - 68ms/epoch - 4ms/step
Epoch 18/50
16/16 - 0s - loss: 133.9499 - val_loss: 118.6777 - 67ms/epoch - 4ms/step
Epoch 19/50
16/16 - 0s - loss: 119.6956 - val_loss: 118.0653 - 70ms/epoch - 4ms/step
Epoch 20/50
16/16 - 0s - loss: 122.3168 - val_loss: 137.6755 - 74

Epoch 5/50
16/16 - 0s - loss: 128.2346 - val_loss: 106.6611 - 80ms/epoch - 5ms/step
Epoch 6/50
16/16 - 0s - loss: 123.1657 - val_loss: 101.7256 - 65ms/epoch - 4ms/step
Epoch 7/50
16/16 - 0s - loss: 105.4199 - val_loss: 117.5505 - 86ms/epoch - 5ms/step
Epoch 8/50
16/16 - 0s - loss: 156.1463 - val_loss: 109.6421 - 81ms/epoch - 5ms/step
Epoch 9/50
16/16 - 0s - loss: 106.7675 - val_loss: 123.9540 - 75ms/epoch - 5ms/step
Epoch 10/50
16/16 - 0s - loss: 114.0172 - val_loss: 103.1073 - 71ms/epoch - 4ms/step
Epoch 11/50
16/16 - 0s - loss: 121.2639 - val_loss: 123.8844 - 77ms/epoch - 5ms/step
Epoch 12/50
16/16 - 0s - loss: 106.2269 - val_loss: 130.1132 - 73ms/epoch - 5ms/step
Epoch 13/50
16/16 - 0s - loss: 110.4520 - val_loss: 105.0443 - 74ms/epoch - 5ms/step
Epoch 14/50
16/16 - 0s - loss: 111.7990 - val_loss: 100.4707 - 79ms/epoch - 5ms/step
Epoch 15/50
16/16 - 0s - loss: 114.2610 - val_loss: 145.2587 - 83ms/epoch - 5ms/step
Epoch 16/50
16/16 - 0s - loss: 115.6010 - val_loss: 102.2456 - 72ms/ep

Epoch 2/50
16/16 - 0s - loss: 92.2460 - val_loss: 106.7784 - 69ms/epoch - 4ms/step
Epoch 3/50
16/16 - 0s - loss: 80.3415 - val_loss: 116.2288 - 69ms/epoch - 4ms/step
Epoch 4/50
16/16 - 0s - loss: 102.7704 - val_loss: 99.1599 - 67ms/epoch - 4ms/step
Epoch 5/50
16/16 - 0s - loss: 93.8336 - val_loss: 87.9054 - 95ms/epoch - 6ms/step
Epoch 6/50
16/16 - 0s - loss: 77.1287 - val_loss: 89.1590 - 79ms/epoch - 5ms/step
Epoch 7/50
16/16 - 0s - loss: 76.9958 - val_loss: 118.2876 - 87ms/epoch - 5ms/step
Epoch 8/50
16/16 - 0s - loss: 80.6474 - val_loss: 92.6565 - 85ms/epoch - 5ms/step
Epoch 9/50
16/16 - 0s - loss: 76.4311 - val_loss: 89.9772 - 69ms/epoch - 4ms/step
Epoch 10/50
16/16 - 0s - loss: 83.3256 - val_loss: 94.5638 - 78ms/epoch - 5ms/step
Epoch 11/50
16/16 - 0s - loss: 76.8394 - val_loss: 122.1153 - 76ms/epoch - 5ms/step
Epoch 12/50
16/16 - 0s - loss: 88.2256 - val_loss: 100.5050 - 69ms/epoch - 4ms/step
Epoch 13/50
16/16 - 0s - loss: 81.0414 - val_loss: 93.1995 - 68ms/epoch - 4ms/step
Epoch 

16/16 - 0s - loss: 80.4923 - val_loss: 88.7351 - 70ms/epoch - 4ms/step
11/11 [==============================] - 0s 2ms/step
Epoch 1/50
16/16 - 0s - loss: 70.6134 - val_loss: 78.7897 - 140ms/epoch - 9ms/step
Epoch 2/50
16/16 - 0s - loss: 70.5038 - val_loss: 85.7074 - 79ms/epoch - 5ms/step
Epoch 3/50
16/16 - 0s - loss: 70.4088 - val_loss: 81.0141 - 86ms/epoch - 5ms/step
Epoch 4/50
16/16 - 0s - loss: 68.2179 - val_loss: 100.1151 - 72ms/epoch - 5ms/step
Epoch 5/50
16/16 - 0s - loss: 78.6781 - val_loss: 77.9532 - 92ms/epoch - 6ms/step
Epoch 6/50
16/16 - 0s - loss: 82.7947 - val_loss: 117.9050 - 84ms/epoch - 5ms/step
Epoch 7/50
16/16 - 0s - loss: 77.4630 - val_loss: 95.8129 - 70ms/epoch - 4ms/step
Epoch 8/50
16/16 - 0s - loss: 81.0929 - val_loss: 80.4815 - 101ms/epoch - 6ms/step
Epoch 9/50
16/16 - 0s - loss: 71.1521 - val_loss: 81.5549 - 75ms/epoch - 5ms/step
Epoch 10/50
16/16 - 0s - loss: 70.2355 - val_loss: 78.7284 - 70ms/epoch - 4ms/step
Epoch 11/50
16/16 - 0s - loss: 68.4084 - val_loss: 

Epoch 48/50
16/16 - 0s - loss: 66.1672 - val_loss: 77.5204 - 75ms/epoch - 5ms/step
Epoch 49/50
16/16 - 0s - loss: 64.5193 - val_loss: 86.6351 - 66ms/epoch - 4ms/step
Epoch 50/50
16/16 - 0s - loss: 73.0269 - val_loss: 72.5114 - 66ms/epoch - 4ms/step
11/11 [==============================] - 0s 2ms/step
Epoch 1/50
16/16 - 0s - loss: 68.7742 - val_loss: 77.2775 - 139ms/epoch - 9ms/step
Epoch 2/50
16/16 - 0s - loss: 67.2070 - val_loss: 77.0922 - 78ms/epoch - 5ms/step
Epoch 3/50
16/16 - 0s - loss: 78.3496 - val_loss: 94.8472 - 85ms/epoch - 5ms/step
Epoch 4/50
16/16 - 0s - loss: 71.9664 - val_loss: 80.1438 - 69ms/epoch - 4ms/step
Epoch 5/50
16/16 - 0s - loss: 74.6487 - val_loss: 85.8079 - 73ms/epoch - 5ms/step
Epoch 6/50
16/16 - 0s - loss: 75.0882 - val_loss: 71.3067 - 83ms/epoch - 5ms/step
Epoch 7/50
16/16 - 0s - loss: 66.7258 - val_loss: 76.1419 - 91ms/epoch - 6ms/step
Epoch 8/50
16/16 - 0s - loss: 64.3293 - val_loss: 88.0674 - 77ms/epoch - 5ms/step
Epoch 9/50
16/16 - 0s - loss: 100.1046 - 

Epoch 46/50
16/16 - 0s - loss: 67.9224 - val_loss: 72.0958 - 94ms/epoch - 6ms/step
Epoch 47/50
16/16 - 0s - loss: 68.0607 - val_loss: 77.6987 - 67ms/epoch - 4ms/step
Epoch 48/50
16/16 - 0s - loss: 82.0213 - val_loss: 82.1009 - 69ms/epoch - 4ms/step
Epoch 49/50
16/16 - 0s - loss: 67.3094 - val_loss: 92.7340 - 70ms/epoch - 4ms/step
Epoch 50/50
16/16 - 0s - loss: 73.6647 - val_loss: 73.6161 - 70ms/epoch - 4ms/step
11/11 [==============================] - 0s 2ms/step
Epoch 1/50
16/16 - 0s - loss: 84.8512 - val_loss: 70.9751 - 140ms/epoch - 9ms/step
Epoch 2/50
16/16 - 0s - loss: 71.9428 - val_loss: 67.8053 - 82ms/epoch - 5ms/step
Epoch 3/50
16/16 - 0s - loss: 64.7466 - val_loss: 68.6366 - 90ms/epoch - 6ms/step
Epoch 4/50
16/16 - 0s - loss: 67.7682 - val_loss: 117.1295 - 77ms/epoch - 5ms/step
Epoch 5/50
16/16 - 0s - loss: 95.0182 - val_loss: 96.3316 - 85ms/epoch - 5ms/step
Epoch 6/50
16/16 - 0s - loss: 70.9327 - val_loss: 82.8134 - 82ms/epoch - 5ms/step
Epoch 7/50
16/16 - 0s - loss: 64.3460 

Epoch 44/50
16/16 - 0s - loss: 61.5845 - val_loss: 70.5963 - 74ms/epoch - 5ms/step
Epoch 45/50
16/16 - 0s - loss: 60.1895 - val_loss: 66.1684 - 72ms/epoch - 5ms/step
Epoch 46/50
16/16 - 0s - loss: 62.6509 - val_loss: 71.2500 - 81ms/epoch - 5ms/step
Epoch 47/50
16/16 - 0s - loss: 73.6360 - val_loss: 77.2179 - 76ms/epoch - 5ms/step
Epoch 48/50
16/16 - 0s - loss: 64.4842 - val_loss: 71.1132 - 82ms/epoch - 5ms/step
Epoch 49/50
16/16 - 0s - loss: 59.3991 - val_loss: 69.1111 - 77ms/epoch - 5ms/step
Epoch 50/50
16/16 - 0s - loss: 58.9559 - val_loss: 79.8860 - 72ms/epoch - 4ms/step
11/11 [==============================] - 0s 2ms/step
Epoch 1/50
16/16 - 0s - loss: 60.3241 - val_loss: 73.0002 - 141ms/epoch - 9ms/step
Epoch 2/50
16/16 - 0s - loss: 63.2111 - val_loss: 75.0384 - 94ms/epoch - 6ms/step
Epoch 3/50
16/16 - 0s - loss: 59.6370 - val_loss: 69.8156 - 100ms/epoch - 6ms/step
Epoch 4/50
16/16 - 0s - loss: 59.7559 - val_loss: 67.7427 - 67ms/epoch - 4ms/step
Epoch 5/50
16/16 - 0s - loss: 58.127

Epoch 42/50
16/16 - 0s - loss: 56.0706 - val_loss: 62.1168 - 81ms/epoch - 5ms/step
Epoch 43/50
16/16 - 0s - loss: 54.0205 - val_loss: 65.1314 - 81ms/epoch - 5ms/step
Epoch 44/50
16/16 - 0s - loss: 56.2006 - val_loss: 73.3554 - 79ms/epoch - 5ms/step
Epoch 45/50
16/16 - 0s - loss: 63.2889 - val_loss: 63.9749 - 71ms/epoch - 4ms/step
Epoch 46/50
16/16 - 0s - loss: 56.0424 - val_loss: 69.3855 - 75ms/epoch - 5ms/step
Epoch 47/50
16/16 - 0s - loss: 63.8705 - val_loss: 71.3429 - 70ms/epoch - 4ms/step
Epoch 48/50
16/16 - 0s - loss: 60.1789 - val_loss: 64.5070 - 69ms/epoch - 4ms/step
Epoch 49/50
16/16 - 0s - loss: 58.6424 - val_loss: 64.0317 - 65ms/epoch - 4ms/step
Epoch 50/50
16/16 - 0s - loss: 64.9082 - val_loss: 65.2585 - 67ms/epoch - 4ms/step
11/11 [==============================] - 0s 2ms/step
Epoch 1/50
16/16 - 0s - loss: 65.4396 - val_loss: 130.3195 - 120ms/epoch - 7ms/step
Epoch 2/50
16/16 - 0s - loss: 90.2535 - val_loss: 78.7788 - 72ms/epoch - 5ms/step
Epoch 3/50
16/16 - 0s - loss: 67.9

Epoch 40/50
16/16 - 0s - loss: 63.5691 - val_loss: 69.7073 - 75ms/epoch - 5ms/step
Epoch 41/50
16/16 - 0s - loss: 67.6858 - val_loss: 60.5502 - 79ms/epoch - 5ms/step
Epoch 42/50
16/16 - 0s - loss: 54.9042 - val_loss: 72.3607 - 87ms/epoch - 5ms/step
Epoch 43/50
16/16 - 0s - loss: 62.7177 - val_loss: 60.8535 - 94ms/epoch - 6ms/step
Epoch 44/50
16/16 - 0s - loss: 57.3705 - val_loss: 56.1174 - 72ms/epoch - 5ms/step
Epoch 45/50
16/16 - 0s - loss: 55.8358 - val_loss: 56.5759 - 72ms/epoch - 5ms/step
Epoch 46/50
16/16 - 0s - loss: 53.8595 - val_loss: 60.0068 - 82ms/epoch - 5ms/step
Epoch 47/50
16/16 - 0s - loss: 52.3138 - val_loss: 55.2691 - 87ms/epoch - 5ms/step
Epoch 48/50
16/16 - 0s - loss: 53.3127 - val_loss: 74.5130 - 74ms/epoch - 5ms/step
Epoch 49/50
16/16 - 0s - loss: 66.9732 - val_loss: 73.9539 - 79ms/epoch - 5ms/step
Epoch 50/50
16/16 - 0s - loss: 55.3320 - val_loss: 80.4311 - 69ms/epoch - 4ms/step
11/11 [==============================] - 0s 2ms/step
Epoch 1/50
16/16 - 0s - loss: 67.0

Epoch 38/50
16/16 - 0s - loss: 49.8472 - val_loss: 64.4781 - 68ms/epoch - 4ms/step
Epoch 39/50
16/16 - 0s - loss: 59.6870 - val_loss: 49.4560 - 83ms/epoch - 5ms/step
Epoch 40/50
16/16 - 0s - loss: 50.9089 - val_loss: 52.8218 - 74ms/epoch - 5ms/step
Epoch 41/50
16/16 - 0s - loss: 49.7042 - val_loss: 53.7298 - 84ms/epoch - 5ms/step
Epoch 42/50
16/16 - 0s - loss: 49.0591 - val_loss: 49.4111 - 71ms/epoch - 4ms/step
Epoch 43/50
16/16 - 0s - loss: 48.5915 - val_loss: 49.5436 - 71ms/epoch - 4ms/step
Epoch 44/50
16/16 - 0s - loss: 49.8333 - val_loss: 49.5549 - 70ms/epoch - 4ms/step
Epoch 45/50
16/16 - 0s - loss: 49.4675 - val_loss: 54.7258 - 66ms/epoch - 4ms/step
Epoch 46/50
16/16 - 0s - loss: 50.9624 - val_loss: 61.3293 - 68ms/epoch - 4ms/step
Epoch 47/50
16/16 - 0s - loss: 53.0134 - val_loss: 53.5849 - 69ms/epoch - 4ms/step
Epoch 48/50
16/16 - 0s - loss: 53.0564 - val_loss: 78.8771 - 72ms/epoch - 4ms/step
Epoch 49/50
16/16 - 0s - loss: 52.9766 - val_loss: 51.2172 - 80ms/epoch - 5ms/step
Epoc

Epoch 36/50
16/16 - 0s - loss: 60.0678 - val_loss: 67.1899 - 90ms/epoch - 6ms/step
Epoch 37/50
16/16 - 0s - loss: 68.4905 - val_loss: 73.4997 - 78ms/epoch - 5ms/step
Epoch 38/50
16/16 - 0s - loss: 77.9627 - val_loss: 68.5695 - 102ms/epoch - 6ms/step
Epoch 39/50
16/16 - 0s - loss: 66.9607 - val_loss: 99.7012 - 77ms/epoch - 5ms/step
Epoch 40/50
16/16 - 0s - loss: 109.8278 - val_loss: 57.9865 - 80ms/epoch - 5ms/step
Epoch 41/50
16/16 - 0s - loss: 55.1596 - val_loss: 71.3170 - 82ms/epoch - 5ms/step
Epoch 42/50
16/16 - 0s - loss: 63.2181 - val_loss: 49.6809 - 75ms/epoch - 5ms/step
Epoch 43/50
16/16 - 0s - loss: 51.1957 - val_loss: 48.4511 - 74ms/epoch - 5ms/step
Epoch 44/50
16/16 - 0s - loss: 50.1308 - val_loss: 49.4640 - 70ms/epoch - 4ms/step
Epoch 45/50
16/16 - 0s - loss: 59.9741 - val_loss: 59.9759 - 94ms/epoch - 6ms/step
Epoch 46/50
16/16 - 0s - loss: 51.7272 - val_loss: 68.9821 - 77ms/epoch - 5ms/step
Epoch 47/50
16/16 - 0s - loss: 56.3880 - val_loss: 48.9787 - 68ms/epoch - 4ms/step
Ep

Epoch 34/50
16/16 - 0s - loss: 45.7808 - val_loss: 48.2375 - 67ms/epoch - 4ms/step
Epoch 35/50
16/16 - 0s - loss: 48.2598 - val_loss: 55.8433 - 68ms/epoch - 4ms/step
Epoch 36/50
16/16 - 0s - loss: 53.7958 - val_loss: 57.3087 - 70ms/epoch - 4ms/step
Epoch 37/50
16/16 - 0s - loss: 57.4595 - val_loss: 51.0876 - 73ms/epoch - 5ms/step
Epoch 38/50
16/16 - 0s - loss: 48.6650 - val_loss: 46.5459 - 71ms/epoch - 4ms/step
Epoch 39/50
16/16 - 0s - loss: 47.4098 - val_loss: 46.2904 - 86ms/epoch - 5ms/step
Epoch 40/50
16/16 - 0s - loss: 47.7876 - val_loss: 95.6212 - 73ms/epoch - 5ms/step
Epoch 41/50
16/16 - 0s - loss: 75.2426 - val_loss: 58.5866 - 81ms/epoch - 5ms/step
Epoch 42/50
16/16 - 0s - loss: 50.5606 - val_loss: 68.7120 - 66ms/epoch - 4ms/step
Epoch 43/50
16/16 - 0s - loss: 65.9124 - val_loss: 70.0383 - 84ms/epoch - 5ms/step
Epoch 44/50
16/16 - 0s - loss: 61.4769 - val_loss: 45.9915 - 73ms/epoch - 5ms/step
Epoch 45/50
16/16 - 0s - loss: 46.8220 - val_loss: 45.7991 - 79ms/epoch - 5ms/step
Epoc

Epoch 32/50
16/16 - 0s - loss: 46.7241 - val_loss: 48.2078 - 77ms/epoch - 5ms/step
Epoch 33/50
16/16 - 0s - loss: 48.8183 - val_loss: 62.3307 - 87ms/epoch - 5ms/step
Epoch 34/50
16/16 - 0s - loss: 63.4079 - val_loss: 46.4769 - 97ms/epoch - 6ms/step
Epoch 35/50
16/16 - 0s - loss: 62.1575 - val_loss: 46.8613 - 75ms/epoch - 5ms/step
Epoch 36/50
16/16 - 0s - loss: 46.9000 - val_loss: 47.6208 - 91ms/epoch - 6ms/step
Epoch 37/50
16/16 - 0s - loss: 47.1479 - val_loss: 45.8300 - 68ms/epoch - 4ms/step
Epoch 38/50
16/16 - 0s - loss: 49.8451 - val_loss: 45.7980 - 73ms/epoch - 5ms/step
Epoch 39/50
16/16 - 0s - loss: 46.7365 - val_loss: 57.5842 - 86ms/epoch - 5ms/step
Epoch 40/50
16/16 - 0s - loss: 59.3430 - val_loss: 67.9694 - 78ms/epoch - 5ms/step
Epoch 41/50
16/16 - 0s - loss: 55.3251 - val_loss: 49.9652 - 73ms/epoch - 5ms/step
Epoch 42/50
16/16 - 0s - loss: 48.4761 - val_loss: 48.3235 - 73ms/epoch - 5ms/step
Epoch 43/50
16/16 - 0s - loss: 49.4073 - val_loss: 55.3880 - 75ms/epoch - 5ms/step
Epoc

Epoch 30/50
16/16 - 0s - loss: 84.6543 - val_loss: 54.1294 - 68ms/epoch - 4ms/step
Epoch 31/50
16/16 - 0s - loss: 59.6436 - val_loss: 57.3379 - 68ms/epoch - 4ms/step
Epoch 32/50
16/16 - 0s - loss: 49.7350 - val_loss: 73.3913 - 65ms/epoch - 4ms/step
Epoch 33/50
16/16 - 0s - loss: 66.3916 - val_loss: 46.6775 - 88ms/epoch - 5ms/step
Epoch 34/50
16/16 - 0s - loss: 58.5788 - val_loss: 47.8607 - 89ms/epoch - 6ms/step
Epoch 35/50
16/16 - 0s - loss: 45.8969 - val_loss: 52.5893 - 71ms/epoch - 4ms/step
Epoch 36/50
16/16 - 0s - loss: 49.9866 - val_loss: 51.4781 - 84ms/epoch - 5ms/step
Epoch 37/50
16/16 - 0s - loss: 57.6902 - val_loss: 48.2590 - 74ms/epoch - 5ms/step
Epoch 38/50
16/16 - 0s - loss: 47.0813 - val_loss: 55.3512 - 77ms/epoch - 5ms/step
Epoch 39/50
16/16 - 0s - loss: 49.7089 - val_loss: 44.6879 - 80ms/epoch - 5ms/step
Epoch 40/50
16/16 - 0s - loss: 46.5810 - val_loss: 44.5390 - 70ms/epoch - 4ms/step
Epoch 41/50
16/16 - 0s - loss: 58.2953 - val_loss: 62.0702 - 80ms/epoch - 5ms/step
Epoc

Epoch 28/50
16/16 - 0s - loss: 48.7918 - val_loss: 74.3517 - 111ms/epoch - 7ms/step
Epoch 29/50
16/16 - 0s - loss: 55.2165 - val_loss: 72.6928 - 75ms/epoch - 5ms/step
Epoch 30/50
16/16 - 0s - loss: 62.6816 - val_loss: 63.0861 - 79ms/epoch - 5ms/step
Epoch 31/50
16/16 - 0s - loss: 57.2010 - val_loss: 47.7412 - 74ms/epoch - 5ms/step
Epoch 32/50
16/16 - 0s - loss: 48.7190 - val_loss: 56.3887 - 69ms/epoch - 4ms/step
Epoch 33/50
16/16 - 0s - loss: 51.9499 - val_loss: 52.3700 - 74ms/epoch - 5ms/step
Epoch 34/50
16/16 - 0s - loss: 46.7551 - val_loss: 45.2816 - 71ms/epoch - 4ms/step
Epoch 35/50
16/16 - 0s - loss: 48.3292 - val_loss: 47.3408 - 68ms/epoch - 4ms/step
Epoch 36/50
16/16 - 0s - loss: 46.9317 - val_loss: 51.0175 - 69ms/epoch - 4ms/step
Epoch 37/50
16/16 - 0s - loss: 45.6704 - val_loss: 46.8325 - 69ms/epoch - 4ms/step
Epoch 38/50
16/16 - 0s - loss: 47.5694 - val_loss: 51.9735 - 82ms/epoch - 5ms/step
Epoch 39/50
16/16 - 0s - loss: 45.8220 - val_loss: 48.2145 - 79ms/epoch - 5ms/step
Epo

Epoch 26/50
16/16 - 0s - loss: 56.6172 - val_loss: 51.6638 - 66ms/epoch - 4ms/step
Epoch 27/50
16/16 - 0s - loss: 45.8139 - val_loss: 79.3355 - 69ms/epoch - 4ms/step
Epoch 28/50
16/16 - 0s - loss: 72.7909 - val_loss: 45.3686 - 76ms/epoch - 5ms/step
Epoch 29/50
16/16 - 0s - loss: 63.0481 - val_loss: 45.4065 - 76ms/epoch - 5ms/step
Epoch 30/50
16/16 - 0s - loss: 58.4652 - val_loss: 52.6353 - 82ms/epoch - 5ms/step
Epoch 31/50
16/16 - 0s - loss: 48.4964 - val_loss: 45.8237 - 94ms/epoch - 6ms/step
Epoch 32/50
16/16 - 0s - loss: 47.9879 - val_loss: 51.2349 - 108ms/epoch - 7ms/step
Epoch 33/50
16/16 - 0s - loss: 59.2976 - val_loss: 52.3963 - 67ms/epoch - 4ms/step
Epoch 34/50
16/16 - 0s - loss: 44.8950 - val_loss: 48.9904 - 75ms/epoch - 5ms/step
Epoch 35/50
16/16 - 0s - loss: 51.9541 - val_loss: 49.1694 - 76ms/epoch - 5ms/step
Epoch 36/50
16/16 - 0s - loss: 47.6345 - val_loss: 49.9459 - 99ms/epoch - 6ms/step
Epoch 37/50
16/16 - 0s - loss: 46.5267 - val_loss: 46.1609 - 70ms/epoch - 4ms/step
Epo

In [60]:
len(MSE_list)

50

In [61]:
MSE_list

[1150.8417103274408,
 684.0531345101512,
 518.4043190005299,
 428.03326337997964,
 347.04554213607304,
 275.6207482961295,
 220.12959285166306,
 197.12311093971357,
 176.6595901319119,
 161.40038364555423,
 154.94140418237782,
 153.99223394659376,
 143.2009389641306,
 137.35267391801614,
 130.12497167354562,
 123.35662234534671,
 115.79472760469011,
 120.9517561636489,
 126.01784789663324,
 120.0870667231211,
 122.92883161087093,
 109.64762665164483,
 111.645383589607,
 98.98975210037126,
 96.92615337889208,
 99.60683155925877,
 88.78346785491252,
 95.01532871153758,
 87.15431380775065,
 84.28981853712554,
 84.93724787796106,
 77.55605422811023,
 74.55763060432336,
 67.25827726708256,
 80.43971452462,
 72.92787841203277,
 71.87332015411509,
 89.49023223976646,
 58.39238654230518,
 53.89101501718782,
 54.39315192538713,
 56.570004270825685,
 75.48884618026445,
 55.72584210183064,
 51.009641799570794,
 56.85873836740708,
 53.26008705389253,
 60.83117202084357,
 53.368256654066755,
 76.28

In [62]:
mean= np.mean(MSE_list)
mean

156.10459219391973

In [63]:
standard_deviation = np.std(MSE_list)
standard_deviation

186.2990185985276

## mean and the standard deviation of the mean squared errors.

In [64]:
print("mean:" , mean)
print("Standard deviation:", standard_deviation)

mean: 156.10459219391973
Standard deviation: 186.2990185985276


# How does the mean of the mean squared errors compare to that from Step B?
## Step B
### After Normlize the data
### mean: 3206.2617439649084
### Standard deviation: 11621.582353270214
## Step D
### mean: 156.10459219391973
### Standard deviation: 186.2990185985276

## The mean of the mean squared errors is much lower compared to that from step B
